pkgs/applications/networking/ipfs-cluster/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, buildGoPackage, fetchFromGitHub, fetchgx, gx-go }:
+
+buildGoPackage rec {
+  name = "ipfs-cluster-${version}";
+  version = "0.5.0";
+  rev = "v${version}";
+
+  goPackagePath = "github.com/ipfs/ipfs-cluster";
+
+  extraSrcPaths = [
+    (fetchgx {
+      inherit name src;
+      sha256 = "0jwz3kd07i5fs0sxds80j8d338skhgxgxra377qxsk0cr2hhj2vm";
+    })
+  ];
+
+  src = fetchFromGitHub {
+    owner = "ipfs";
+    repo = "ipfs-cluster";
+    inherit rev;
+    sha256 = "132whjyplcifq8747hcdrgbc0amhp618dg049jq5nyslcxfgdypm";
+  };
+
+  preBuild = ''
+    # fetchgx stores packages by their ipfs hash
+    # this will rewrite github.com/ imports to gx/ipfs/
+    cd go/src/${goPackagePath}
+    ${gx-go}/bin/gx-go rewrite
```

Comment:
This reference is incorrect when cross-building because we need `gx-go` to run during build. 
Instead, please add `nativeBuildInputs = [ gx-go ];` above and just write `gx-go` here (no absolute path).

Reply:
Thank you for your help @xeji , addressed in 314d1fcdcb5b8602d634beaf577757ee5e8b6aee